File: Dataset/Leetcode/test/102/76.cpp
class Solution {
public:
    vector<vector<int>> XXX(TreeNode* root) {
        vector<vector<int>> bfs; 
        vector<int> temp; 
        TreeNode* p; 
        queue<TreeNode*> q;
        if(!root) return bfs;
        q.push(root);
        TreeNode* last = root;
        while(!q.empty()){
            p = q.front();
            q.pop();
            temp.push_back(p->val);
            if(p->left)
                q.push(p->left);
            if(p->right)
                q.push(p->right);
            if(p == last){
                bfs.push_back(temp);
                last = q.back();
                temp.clear();
            }
        }
        return bfs;
    }
};

File: Dataset/Leetcode/test/112/76.py
class Solution(object):
    def XXX(self, root, targetSum):
        """
        :type root: TreeNode
        :type targetSum: int
        :rtype: bool
        """
        #递归
        if not root:
            return False
        elif not root.left and not root.right: #root是否为叶子节点
            return root.val == targetSum
        return self.XXX(root.left,targetSum-root.val) or self.XXX(root.right,targetSum-root.val)

